File: exercism/javascript/strain/strain.js

const keep = (xs, f) =>
  xs.reduce((memo, x) => f(x) ? (memo.push(x) && memo) : memo , [])

const discard = (xs, f) =>
  keep(xs, (x) => !f(x))

module.exports = { keep, discard }

File: exercism/csharp/strain/Strain.cs
using System;
using System.Collections.Generic;

public static class EnumerableExtensions
{
    public static IEnumerable<T> Keep<T> (this IEnumerable<T> source, Func<T, bool> f)
    {
        foreach (var item in source) if (f(item)) yield return item;
    }

    public static IEnumerable<T> Discard<T> (this IEnumerable<T> source, Func<T, bool> f)
    {
        return source.Keep(x => !f(x));
    }
}
